### PR TITLE
integration: fix resultCount types to not overflow, update expected vals.

### PR DESCRIPTION
**Pull request by cpu**
#### integration: fix resultCount types to not overflow

Using `uint8` as the type for the count fields in the `resultCount` type produces overflows if more than 255 certificates with a given result level are linted.

Our integration test corpus is just shy of 600,000 certificates so `uint32` should be more than sufficient.

####  integration: update expected values to correct overflows.

Any lints that had more than 255 results at a given level will have overflowed, meaning expected counts were not correct. Also updates the expected values for the small config (which had drifted out of sync).

Resolves https://github.com/zmap/zlint/issues/555

### CODE REVIEW

File: v3/integration/result.go [hunk 1]
```diff
@@ -9,10 +9,10 @@ import (
 )
 
 type resultCount struct {
-	FatalCount  uint8 `json:",omitempty"`
-	ErrCount    uint8 `json:",omitempty"`
-	WarnCount   uint8 `json:",omitempty"`
-	NoticeCount uint8 `json:",omitempty"`
+	FatalCount  uint32 `json:",omitempty"`
```
Comment (christopher-henderson) [inline v3/integration/result.go:12]: image

I'm gonna pull this branch and give it a quick smoke check myself as well as double check the sanity if the new values.

Comment (cpu) [reply v3/integration/result.go:12]: Thank you! Take your time :pray:

**Review by christopher-henderson — APPROVED** [review]
A quick spot check that the new values make sense in that they would overflow to the old values.

```go
package main

func main() {
	var overflows = [][2]uint32{
		{322, 66},
		{431, 175},
		{738, 226},
		{3463, 135},
		{1677, 141},
		{607, 95},
		{607, 95},
		{365, 109},
		{325, 69},
		{724, 212},
		{459, 203},
		{705, 193},
		{678, 166},
		{593196, 44},
		{4680, 72},
		{339, 83},
		{284, 28},
		{7191, 23},
		{9872, 144},
		{16755, 115},
		{59194, 58},
		{816, 48},
		{574, 62},
		{33636, 100},
		{2025, 233},
		{19776, 64},
		{329, 73},
		{552, 40},
		{1944, 152},
		{1122, 98},
	}
	for _, pair := range overflows {
		is := pair[0]
		was := pair[1]
		if uint32(uint8(is)) != was {
			panic("💣")
		}
	}
}
```

File: v3/integration/config.json [hunk 2]
```diff
@@ -638,47 +647,47 @@
...
     "n_sub_ca_eku_missing": {
-      "NoticeCount": 166
+      "NoticeCount": 678
     },
     "n_sub_ca_eku_not_technically_constrained": {
       "NoticeCount": 10
     },
     "n_subject_common_name_included": {
-      "NoticeCount": 44
+      "NoticeCount": 593196
```
Comment (christopher-henderson) [inline v3/integration/config.json:662]: This...is especially terrifying.

**Comment by zakird** [comment]
Well, everything about this ticket makes me sad, but I'm glad we're getting it cleaned up.

**Review by zakird — APPROVED** [review]

File: v3/integration/config.json (hunk 2 above)
Comment (cpu) [reply v3/integration/config.json:662]: but expected! That's why it's a Notice level. The deprecation of subject CNs is... long and ongoing. Honestly seeing only 44 notices in our corpus before for this lint should have been a red flag :sweat:

**Pull request merged**